File: objc/State/State/PenTool.m
#import "PenTool.h"

@implementation PenTool {
    Point sPoint, ePoint;
    BOOL mouseDown;
}

-(instancetype)init {
    if(self = [super init]) {
        mouseDown = NO;
    }
    return self;
}

-(void)mouseTo:(Point) p {
    if(mouseDown) {
        NSLog(@"f");
    }
}

-(void)mouseDown:(Point) p {
    sPoint = p;
    mouseDown = YES;
}

-(void)mouseUp:(Point) p {
    ePoint = p;
    mouseDown = NO;
}

@end

File: swift/State/State/PenTool.swift
import Foundation

class PenTool : AbstractTool {
    var sPoint: Point?
    var ePoint: Point?
    var mouseDown: Bool = false
    
    init() {
        
    }
    
    func mouseTo(_ point: Point) -> Void {
        if mouseDown {
            print("Print a LINE from start point to current point")
        }
    }
    
    func mouseDown(_ point: Point) -> Void {
        sPoint = point
        mouseDown = true
    }
    
    func mouseUp(_ point: Point) -> Void {
        ePoint = point
        mouseDown = false
    }
}
